cmd/soroban-rpc/internal/integrationtest/infrastructure/test.go

```diff
@@ -0,0 +1,717 @@
+	i.runSuccessfulComposeCommand(upCmd...)
+	if i.runRPCInContainer() {
+		i.rpcContainerLogsCommand = i.getComposeCommand("logs", "--no-log-prefix", "-f", "rpc")
+		writer := testLogWriter{t: i.t, prefix: fmt.Sprintf(`rpc="container" version="%s" `, i.rpcContainerVersion)}
+		i.rpcContainerLogsCommand.Stdout = writer
+		i.rpcContainerLogsCommand.Stderr = writer
+		require.NoError(i.t, i.rpcContainerLogsCommand.Start())
+	}
+	i.fillContainerPorts()
+}
+
+func (i *Test) stopContainers() {
+	// There were containerized workloads we should bring down
+	downCmd := []string{"down"}
+	if i.runRPCInContainer() && i.onlyRPC {
+		downCmd = append(downCmd, "rpc")
+	}
+	downCmd = append(downCmd, "-v")
+	i.runSuccessfulComposeCommand(downCmd...)
+
+}
+
+func (i *Test) GetPorts() TestPorts {
+	return i.testPorts
+}
+
+func (i *Test) runRPCInContainer() bool {
+	return i.rpcContainerVersion != ""
+}
+
+func (i *Test) GetRPCLient() *Client {
+	return i.rpcClient
+}
+func (i *Test) MasterKey() *keypair.Full {
+	return keypair.Root(StandaloneNetworkPassphrase)
+}
+
+func (i *Test) MasterAccount() txnbuild.Account {
+	return i.masterAccount
+}
+
+func (i *Test) GetSorobanRPCURL() string {
+	return fmt.Sprintf("http://localhost:%d", i.testPorts.RPCPort)
+}
+
+func (i *Test) GetAdminURL() string {
+	return fmt.Sprintf("http://localhost:%d", i.testPorts.RPCAdminPort)
+}
+
+func (i *Test) getCoreInfo() (*proto.InfoResponse, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second)
+	defer cancel()
+	return i.coreClient.Info(ctx)
+}
+
+func (i *Test) waitForCheckpoint() {
+	i.t.Log("Waiting for checkpoint...")
+	require.Eventually(i.t,
+		func() bool {
+			info, err := i.getCoreInfo()
+			return err == nil && info.Info.Ledger.Num > checkpointFrequency
+		},
+		30*time.Second,
+		time.Second,
+	)
+}
+
+func (i *Test) getRPConfigForContainer() rpcConfig {
+	return rpcConfig{
+		// The container needs to listen on all interfaces, not just localhost
+		// (otherwise it can't be accessible from the outside)
+		endPoint:       fmt.Sprintf("0.0.0.0:%d", inContainerRPCPort),
+		adminEndpoint:  fmt.Sprintf("0.0.0.0:%d", inContainerRPCAdminPort),
+		stellarCoreURL: fmt.Sprintf("http://%s:%d", inContainerCoreHostname, inContainerCoreHTTPPort),
+		// Container's default path to captive core
+		coreBinaryPath: "/usr/bin/stellar-core",
+		// The file will be inside the container
+		captiveCoreConfigPath: "/stellar-core.cfg",
+		// Any writable directory would do
+		captiveCoreStoragePath: "/tmp/captive-core",
+		archiveURL:             fmt.Sprintf("http://%s:%d", inContainerCoreHostname, inContainerCoreArchivePort),
+		sqlitePath:             "/db/" + filepath.Base(i.sqlitePath),
+	}
+}
+
+func (i *Test) getRPConfigForDaemon() rpcConfig {
+	coreBinaryPath := os.Getenv("SOROBAN_RPC_INTEGRATION_TESTS_CAPTIVE_CORE_BIN")
+	if coreBinaryPath == "" {
+		i.t.Fatal("missing SOROBAN_RPC_INTEGRATION_TESTS_CAPTIVE_CORE_BIN")
```

Comment:
Can we check the PATH by default? (e.g. `which stellar-core`) It's always mildly annoying to provide this

Reply:
We can do it separately from this change (that check predates the PR)

(I never have it in my path though :)